lib/checks/canFindDenominatorInNumerator.js

```diff
@@ -19,17 +20,42 @@ function canFindDenominatorInNumerator(node) {
   if (Node.Type.isParenthesis(denominator)) {
     denominator = denominator.content;
   }
+  if (!(numerator.op === '+' || numerator.op === '-' ||
+        denominator.op === '+' || numerator.op === '-')) {
+    return false;
+  }
+  if (denominator.op !== '+') {
+    return false;
+  }
+
+  let numeratorFirstTerm;
+  if (numerator.op === '+') {
+    numeratorFirstTerm = new Node.PolynomialTerm(numerator.args[0]);
```

Comment:
Yeah, this is what I wanted to talk about actually. One of the main issues is that it's not sorted, so I wanted to ask you all if we should add a sorting function which will run before this. Or if you don't see any merit in that, we could add some logic to find the highest polynomial term right into the function (or outside of it if you want).